src/databricks/labs/ucx/install.py

```diff
@@ -819,23 +828,37 @@ def _job_clusters(self, names: set[str]):
             )
         return clusters
 
-    def _cluster_node_type(self, spec: compute.ClusterSpec) -> compute.ClusterSpec:
-        cfg = self.current_config
-        valid_node_type = False
-        if cfg.custom_cluster_policy_id is not None:
-            if self._check_policy_has_instance_pool(cfg.custom_cluster_policy_id):
-                valid_node_type = True
-        if not valid_node_type:
-            if cfg.instance_pool_id is not None:
-                return replace(spec, instance_pool_id=cfg.instance_pool_id)
-            spec = replace(spec, node_type_id=self._ws.clusters.select_node_type(local_disk=True))
+    def _cluster_policy_definition(self, conf: dict, instance_profile: str) -> str:
+        policy_definition = {
+            "spark_version": {"type": "fixed", "value": self._ws.clusters.select_spark_version(latest=True)},
+            "node_type_id": {"type": "fixed", "value": self._ws.clusters.select_node_type(local_disk=True)},
+        }
+        if conf:
+            for key, value in conf.items():
+                if key.startswith("spark.hadoop.javax.jdo.option") or key.startswith("sql.hive.metastore"):
+                    policy_definition[f"spark_conf.{key}"] = {"type": "fixed", "value": value}
+                elif self._ws.config.is_aws and key.startswith("spark.databricks.hive.metastore.glueCatalog.enabled"):
+                    policy_definition[f"spark_conf.{key}"] = {"type": "fixed", "value": value}
+                else:
+                    continue
```

Comment:
why continue? loosing data, no?

Reply:
not losing data. the spark conf contains only spark.hadoop.javax.jdo.option or databricks.hive.metastore.glueCatalog in the function _get_ext_hms_conf_from_policy so else statement shouldn't have any